clash-ghc/src-ghc/Clash/GHC/LoadModules.hs

```diff
@@ -521,21 +522,10 @@ unpackKnownConf ty
   | otherwise
   = error $ $(curLoc) ++ "Could not unpack domain configuration."
  where
-  asActiveEdge "Rising"  = Rising
-  asActiveEdge "Falling" = Falling
-  asActiveEdge x = error $ $(curLoc) ++ "Unknown active edge: " ++ show x
-
-  asResetKind "Synchronous"  = Synchronous
-  asResetKind "Asynchronous" = Asynchronous
-  asResetKind x = error $ $(curLoc) ++ "Unknown reset kind: " ++ show x
-
-  asInitBehaviour "Unknown" = Unknown
-  asInitBehaviour "Defined" = Defined
-  asInitBehaviour x = error $ $(curLoc) ++ "Unknown init behaviour: " ++ show x
-
-  asResetPolarity "ActiveHigh" = ActiveHigh
-  asResetPolarity "ActiveLow"  = ActiveLow
-  asResetPolarity x = error $ $(curLoc) ++ "Unknown reset polarity: " ++ show x
+  asActiveEdge x = fromMaybe (error $ $(curLoc) ++ "Unknown active edge: " ++ show x) (readMaybe x)
+  asResetKind x = fromMaybe (error $ $(curLoc) ++ "Unknown reset kind: " ++ show x) (readMaybe x)
+  asInitBehaviour x = fromMaybe (error $ $(curLoc) ++ "Unknown init behaviour: " ++ show x) (readMaybe x)
```

Comment:
Ofc it shouldn't happen, but this might end up at a user's screen. Hence, we should use :us: spelling.

Reply:
As this function is partial, I think it should have a `HasCallStack` constraint.